addons/mrp_byproduct/models/mrp_production.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from openerp import api, models, _
+
+
+class MrpProduction(models.Model):
+    _inherit = 'mrp.production'
+
+    @api.multi
+    def action_confirm(self):
+        StockMove = self.env['stock.move']
```

Comment:
If you assign variable then put into loop or else directly put into loop both are same, @mpa-odoo  you are kidding with this comment :question: